Backend/starter_project_zsilentcoders/BlogApp3/BlogApp.Persistence/Repositories/ReviewRepository.cs

```diff
@@ -0,0 +1,34 @@
+using BlogApp.Application.Contracts.Persistence;
+using BlogApp.Domain;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace BlogApp.Persistence.Repositories
+{
+    public class ReviewRepository: GenericRepository<Review>, IReviewRepository
+    {
+        private readonly BlogAppDbContext _dbContext;
+
+        public ReviewRepository(BlogAppDbContext dbContext) : base(dbContext)
+        {
+            _dbContext = dbContext;
+        }
+        public async Task ChangeApprovalStatus(Review review, bool? IsResolved)
+        {
+            review.IsResolved = IsResolved;
+            _dbContext.Entry(review).State = EntityState.Modified;
+            await _dbContext.SaveChangesAsync();
```

Comment:
This(SaveChangesAsync) will be called by UnitOfWork if you are using UnitOfWork to access this repo. But make sure to call the unitOfWork.save() method after using the unitOfWork in you handdler.